backend/pkg/httpserver/update_saved_search.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpserver
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"net/http"
+	"strings"
+
+	"github.com/GoogleChrome/webstatus.dev/lib/gcpspanner/spanneradapters/backendtypes"
+	"github.com/GoogleChrome/webstatus.dev/lib/gen/openapi/backend"
+	"github.com/GoogleChrome/webstatus.dev/lib/httpmiddlewares"
+)
+
+// validateSavedSearchUpdate handles validation when updating an existing SavedSearch.
+func validateSavedSearchUpdate(input *backend.UpdateSavedSearchJSONRequestBody) *fieldValidationErrors {
+	activeMasks := make(map[backend.SavedSearchUpdateRequestUpdateMask]bool)
+	var invalidMasks []string
+
+	fieldErrors := &fieldValidationErrors{fieldErrorMap: nil}
+
+	if len(input.UpdateMask) == 0 {
+		fieldErrors.addFieldError("update_mask", errors.New("update_mask must be set"))
+
+		return fieldErrors
+	}
+
+	for _, mask := range input.UpdateMask {
+		switch mask {
+		case
+			backend.SavedSearchUpdateRequestMaskName,
+			backend.SavedSearchUpdateRequestMaskQuery,
+			backend.SavedSearchUpdateRequestMaskDescription:
+			activeMasks[mask] = true
+		default:
+			invalidMasks = append(invalidMasks, string(mask))
+		}
+	}
+
+	if len(invalidMasks) > 0 {
+		fieldErrors.addFieldError("update_mask", errors.New("invalid update_mask values: "+
+			strings.Join(invalidMasks, ", "),
+		))
+
+		return fieldErrors
+	}
```

Comment:
note: we only return early here and the other update mask check above because we need the update mask to know which fields to validate